aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -94,22 +72,28 @@ static void checkCuda90Bug(int i_m, int i_n, int i_k)
 
 Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& mat1, const Tensor& mat2, Scalar beta, Scalar alpha) {
   TORCH_CHECK(
-    (mat1.dim() == 2) && (mat2.dim() == 2) &&
-    (self.dim() == 2) && (result.dim() == 2),
+    (mat1.dim() == 2) && (mat2.dim() == 2),
```

Comment:
Don't you have to check self and result too? Or is that done later on somewhere?

Reply:
I have just added those checks below